packages/lambda-xyz/src/tile.set.cache.ts

```diff
@@ -9,20 +16,115 @@ export function getTileSet(name: string, projection: Epsg): TileSet | undefined
+    copy.extentOverride = Bounds.fromJson(image.imagery.bounds);
+
+    const rule = {
+        id,
+        minZoom: 0,
+        maxZoom: image.rule.maxZoom,
+        priority: 0,
+    };
+    copy.tileSet.imagery = { [id]: rule };
+
+    copy.imagery = [
+        {
+            rule,
+            imagery: image.imagery,
+        },
+    ];
+
+    return copy;
+}
+
+export async function loadTileSet(name: string, projection: Epsg): Promise<TileSet | null> {
+    const subsetIndex = name.indexOf(':');
+    const subsetName = subsetIndex == -1 ? '' : name.slice(subsetIndex + 1);
+    if (subsetName !== '') {
+        name = name.slice(0, subsetIndex);
+    }
     let tileSet = getTileSet(name, projection);
     if (tileSet == null) {
         tileSet = new TileSet(name, projection);
         TileSets.set(tileSet.id, tileSet);
     }
-    req.timer.start('tileset:load');
     const loaded = await tileSet.load();
-    req.timer.end('tileset:load');
     if (!loaded) {
         TileSets.delete(tileSet.id);
         return null;
     }
+    if (subsetName !== '') {
+        const image = tileSet.imagery.find((i) => i.imagery.name === subsetName);
+        if (image == null) return null;
+        const subTileSet = individualTileSet(tileSet, image);
+        subTileSet.tileSet.background = undefined;
+        return subTileSet;
+    }
     return tileSet;
 }
+
+async function collectTileSets(tileSets: TileSet[], name: string, projection: Epsg | null): Promise<void> {
+    if (projection == null) {
+        for (const code of ProjectionTileMatrixSet.targetCodes()) {
+            const epsg = Epsg.get(code);
+            await collectTileSets(tileSets, name, epsg);
+        }
+        return;
+    }
+
+    const parent = await loadTileSet(name, projection);
+    if (parent != null) {
+        tileSets.push(parent);
+        if (parent.imagery != null) {
+            for (const image of parent.imagery) {
+                tileSets.push(individualTileSet(parent, image, name + ':' + image.imagery.name));
+            }
+        }
+    }
+}
+
+function compareByTitle(a: TileSet, b: TileSet): number {
+    return a.title.localeCompare(b.title);
+}
+
+/**
+ * Load a collection of TileSets. Used by WMTSCapabilities
+
+ * @param name if an empty string load all TileSets
+ * @param projection if null load all projections
+ */
+export async function loadTileSets(nameStr: string, projection: Epsg | null): Promise<TileSet[]> {
+    if (nameStr !== '' && nameStr[0] !== '@' && projection != null) {
+        // single tileSet
+        const ts = await loadTileSet(nameStr, projection);
+
+        return ts == null ? [] : [ts];
+    }
+
+    const { name, tag } = Aws.tileMetadata.TileSet.parse(nameStr);
+
+    const tileSets: TileSet[] = [];
+
+    if (name === '') {
+        for (const tsn of TileSetNameValues()) {
+            await collectTileSets(tileSets, tag == null ? tsn : `${tsn}@${tag}`, projection);
```

Comment:
I think we need to multithread this, otherwise this is doing upto `TileSetNames.length x projection.length` queries in series.

if we add for instance CITM projection + Topo 50 & 250 could take a while